src/components/forms/FileInput/FileInput.tsx

```diff
@@ -12,164 +12,186 @@ interface FileInputProps {
-  name,
-  id,
-  disabled,
-  multiple,
-  className,
-  accept,
-  onChange,
-  onDrop,
-  inputRef,
-  ...inputProps
-}: FileInputProps & JSX.IntrinsicElements['input']): React.ReactElement => {
-  const [isDragging, setIsDragging] = useState(false)
-  const [showError, setShowError] = useState(false)
-  const [files, setFiles] = useState<FileList | null>(null)
-
-  const fileInputClasses = classnames(
-    'usa-file-input',
-    {
-      'usa-file-input--disabled': disabled,
-    },
-    className
-  )
-
-  const targetClasses = classnames('usa-file-input__target', {
-    'usa-file-input--drag': isDragging,
-    'has-invalid-file': showError,
-  })
-
-  const hideDragText =
-    /rv:11.0/i.test(navigator.userAgent) ||
-    /Edge\/\d./i.test(navigator.userAgent)
-
-  const dragText = multiple ? 'Drag files here or ' : 'Drag file here or '
-
-  const filePreviews = []
-  if (files) {
-    for (let i = 0; i < files?.length; i++) {
-      const imageId = makeSafeForID(files[parseInt(`${i}`)].name)
-      const key = `filePreview_${imageId}`
-      filePreviews.push(
-        <FilePreview
-          key={key}
-          imageId={imageId}
-          file={files[parseInt(`${i}`)]}
-        />
-      )
-    }
-  }
+export type FileInputRef = {
+  clearFiles: () => void
+  input: HTMLInputElement | null
+  files: FileList | []
+}
 
-  const instructionClasses = classnames('usa-file-input__instructions', {
-    'display-none': filePreviews.length > 0,
-  })
-
-  const previewHeaderText =
-    filePreviews.length > 1
-      ? `${filePreviews.length} files selected`
-      : 'Selected file'
-
-  const preventInvalidFiles = (e: React.DragEvent): void => {
-    setShowError(false)
-
-    if (accept) {
-      const acceptedTypes = accept.split(',')
-      let allFilesAllowed = true
-      for (let i = 0; i < e.dataTransfer.files.length; i += 1) {
-        const file = e.dataTransfer.files[parseInt(`${i}`)]
-        if (allFilesAllowed) {
-          for (let j = 0; j < acceptedTypes.length; j += 1) {
-            const fileType = acceptedTypes[parseInt(`${j}`)]
-            allFilesAllowed =
-              file.name.indexOf(fileType) > 0 ||
-              file.type.includes(fileType.replace(/\*/g, ''))
-            if (allFilesAllowed) break
-          }
-        } else break
+export const FileInput = forwardRef(
+  (
+    {
+      name,
+      id,
+      disabled,
+      multiple,
+      className,
+      accept,
+      onChange,
+      onDrop,
+      ...inputProps
+    }: FileInputProps & JSX.IntrinsicElements['input'],
+    ref: React.Ref<FileInputRef | null>
+  ): React.ReactElement => {
+    const internalRef = useRef<HTMLInputElement>(null)
+    const [isDragging, setIsDragging] = useState(false)
+    const [showError, setShowError] = useState(false)
+    const [files, setFiles] = useState<FileList | []>([])
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        input: internalRef.current,
+        clearFiles: (): void => setFiles([]),
```

Comment:
🤔 Should this also surface a value that says if the input has an error? Right now we manually duplicate code from `preventInValidFiles` for determining if the input has an error when invalid files for `accept` were dropped. Maybe there should be the option to say `fileInputRef.current.hasErrors` ? 

This matter because on those cases we don't want to clear the input or handle our async vents - we want to keep it in the invalid state and show the built-in file input errors for `showError`.